src/appshell/view/preferences/appearancepreferencesmodel.cpp

```diff
@@ -78,6 +93,51 @@ QStringList AppearancePreferencesModel::accentColors() const
     return uiConfiguration()->possibleAccentColors();
 }
 
+void AppearancePreferencesModel::resetThemeToDefault()
+{
+    uiConfiguration()->resetCurrentThemeToDefault(currentTheme().codeKey);
+    notationConfiguration()->resetCurrentBackgroundColorToDefault();
+    emit backgroundColorChanged();
+    emit themesChanged();
+}
+
+bool AppearancePreferencesModel::enableHighContrastChecked()
```

Comment:
I believe you can use isCurrentThemeHighContrast directly in qml (after marking it as Q_INVOKABLE). You can remove  enableHighContrastChecked after those changes

Reply:
I have removed `isCurrentThemeHighContrast` from `appearancepreferencesmodel.cpp` completely, as suggested. `isCurrentThemeHighContrast` and `isCurrentThemeGeneral` now lie in `uiconfiguration.cpp`. This is what `enableHighContrastChecked()` looks like now:
```c++
bool AppearancePreferencesModel::enableHighContrastChecked()
{
    return uiConfiguration()->isCurrentThemeHighContrast();
}
```
I need the Q_INVOKABLE function (i.e. `enableHighContrastChecked()`) inside AppearancePreferencesModel class for it to work, which is why i can't remove it from here. (unless there is a way to use Q_INVOKABLE functions from different files?)

From what I understand, `uiconfiguration.cpp` file represents the lowest "ModelData" layer, and it sends data up to `uitheme.cpp` and other files i.e. the "ViewModel" layer. I think using a function directly in QML from `uiconfiguration.cpp` will mess up this layering system.

Reply:
>From what I understand, uiconfiguration.cpp file represents the lowest "ModelData" layer, and it sends data up to uitheme.cpp and other files i.e. the "ViewModel" layer. I think using a function directly in QML from uiconfiguration.cpp will mess up this layering system.


Yes, you are absolutely right :) 

There were 2 different isCurrentThemeHighContrast (one in the model, and another in the configuration) and separately enableHighContrastChecked that just called isCurrentThemeHighContrast of the model

Your suggested solution is good: we will have Q_INVOKABLE enableHighContrastChecked in the model and isCurrentThemeHighContrast in the configuration

Reply:
Understood!
I've made the appropriate changes :)